plugins/codescene/src/components/CodeSceneProjectDetailsPage/CodeSceneProjectDetailsPage.test.tsx

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React from 'react';
+import { CodeSceneProjectDetailsPage } from './CodeSceneProjectDetailsPage';
+import { ThemeProvider } from '@material-ui/core';
+import { lightTheme } from '@backstage/theme';
+import { rest } from 'msw';
+import { setupServer } from 'msw/node';
+import {
+  setupRequestMockHandlers,
+  renderInTestApp,
+  TestApiRegistry,
+} from '@backstage/test-utils';
+import { CodeSceneApi, codesceneApiRef } from '../../api/api';
+import { ApiProvider } from '@backstage/core-app-api';
+import { Analysis } from '../../api/types';
+import { ConfigReader } from '@backstage/config';
+import { configApiRef } from '@backstage/core-plugin-api';
+
+jest.mock('react-router-dom', () => {
```

Comment:
Probably better to render on a particular route here instead if possible. Should be a bit easier to maintain and probably more accurate. I don't think you actually get a number back here in reality right?

Reply:
@Rugvip Thanks for the review. Can you clarify this one point though? I took inspiration from other test files in the repository to write this one. What do you mean with "render a particular route"?

Also, the "number we get back" is a hardcoded `projectId` to later perform assertions through the tests. Is there a better/alternative way to do this?

Reply:
Something like this https://github.com/backstage/backstage/blob/ef889d389b6c06ba19afcfe405f699301c291b22/plugins/xcmetrics/src/components/BuildsPage/BuildsPage.test.tsx#L32

Reply:
@julioz pinging a bit about this one - I agree that this probably should be updated

Reply:
Hi, yes I will pick this up - got a bit busy with other work stuff but will come back to this branch soon 👍